src/main/include/controllers/DrivetrainController.hpp

```diff
@@ -0,0 +1,341 @@
+     */
+    Eigen::Matrix<double, 2, 1> GetInputs() const;
+
+    /**
+     * Returns the currently set local outputs.
+     *
+     * heading, left position, left velocity, right position,
+     * right velocity, and angular velocity.
+     */
+    const Eigen::Matrix<double, 3, 1>& GetOutputs() const;
+
+    /**
+     * Returns the estimated outputs based on the current state estimate.
+     *
+     * This provides only local measurements.
+     */
+    Eigen::Matrix<double, 3, 1> EstimatedLocalOutputs() const;
+
+    /**
+     * Returns the estimated outputs based on the current state estimate.
+     *
+     * This provides global measurements (including pose).
+     */
+    Eigen::Matrix<double, 6, 1> EstimatedGlobalOutputs() const;
+
+    /**
+     * Executes the control loop for a cycle.
+     *
+     * @param dt Timestep between each Update() call
+     */
+    void Update(units::second_t dt, units::second_t elapsedTime);
+
+    /**
+     * Resets any internal state.
+     */
+    void Reset();
+
+    /**
+     * Resets any internal state.
+     *
+     * @param initialPose Initial pose for state estimate.
+     */
+    void Reset(const frc::Pose2d& initialPose);
+
+    Eigen::Matrix<double, 2, 1> Controller(
+        const Eigen::Matrix<double, 10, 1>& x,
+        const Eigen::Matrix<double, 5, 1>& r);
+
+    static Eigen::Matrix<double, 10, 1> Dynamics(
+        const Eigen::Matrix<double, 10, 1>& x,
+        const Eigen::Matrix<double, 2, 1>& u);
+
+    static Eigen::Matrix<double, 3, 1> LocalMeasurementModel(
+        const Eigen::Matrix<double, 10, 1>& x,
+        const Eigen::Matrix<double, 2, 1>& u);
+
+    static Eigen::Matrix<double, 6, 1> GlobalMeasurementModel(
+        const Eigen::Matrix<double, 10, 1>& x,
+        const Eigen::Matrix<double, 2, 1>& u);
+
+    class State {
+    public:
+        static constexpr int kX = 0;
+        static constexpr int kY = 1;
+        static constexpr int kHeading = 2;
+        static constexpr int kLeftVelocity = 3;
+        static constexpr int kRightVelocity = 4;
+        static constexpr int kLeftPosition = 5;
+        static constexpr int kRightPosition = 6;
+        static constexpr int kLeftVoltageError = 7;
+        static constexpr int kRightVoltageError = 8;
+        static constexpr int kAngularVelocityError = 9;
+    };
+
+    class Input {
+    public:
+        static constexpr int kLeftVoltage = 0;
+        static constexpr int kRightVoltage = 1;
+    };
+
+    class LocalOutput {
+    public:
+        static constexpr int kHeading = 0;
+        static constexpr int kLeftPosition = 1;
+        static constexpr int kRightPosition = 2;
+    };
+
+    class GlobalOutput {
+    public:
+        static constexpr int kX = 0;
+        static constexpr int kY = 1;
+        static constexpr int kHeading = 2;
+        static constexpr int kLeftPosition = 3;
+        static constexpr int kRightPosition = 4;
+        static constexpr int kAngularVelocity = 5;
+    };
```

Comment:
These can go above the function definitions. The class order is generally: classes/enums, static constant variables, member variables, functions. That repeats for each access specifier, which go: public, protected, private (they are ommitted if they are empty).